### DISCUSSION TITLE
Slow page load with SSG

### DISCUSSION THREAD

**Discussion by michaelketzer** [vercel/next.js#22104] [Help] [answered]
Hey,

i have a wesbite under the following domain https://dota2.shokz.tv/

On the top menu there is an option "Alle Artikel", clicking this link causing the URL to switch immediately but the page switches after ~ 5 seconds.
I am using SSG and loading all the articles in getStaticProps:
https://github.com/michaelketzer/shokztv-web_v3/blob/main/pages/articles.tsx

The only thing I have in mind is probably the case, that i'm loading already the whole content of the articles which kinda causing the json that is preloaded to be bigger. 

Is this the reason? I mean, i have only around 66 articles and this seems really an issue if it is that slow.

**Comment by nicosh** [comment]
If the payload of each article is too heavy that can be the issue.
Since you are using `<Link />` you can show a spinner until the page is loaded. However i think you should use ISR  with `fallback : true` and your page will be fast AF

**Reply by michaelketzer** [reply]
Thanks for your reply, i will give it a try.

**Comment by mkrtchian** [comment, marked as answer]
Your problem comes actually from your usage of `react-lines-ellipsis`. There is a [known limitation of that library about rendering the page on server side](https://github.com/xiaody/react-lines-ellipsis/issues/12#issuecomment-355515160): it needs to do some heavy calculation on client side to truncate the exact amount of text according to the client display size, so that the indicated number of lines to keep is met.

I have forked your repo and made another small component that does not use `react-lines-ellipsis`, but does the truncation by hand in a way to avoid doing calculations on client side. [The commit is here](https://github.com/mkrtchian/shokztv-web_v3/commit/aebdfa532a32ca8e7c2733c093092420f8ed133b).

The disadvantage is that I cut an arbitrary amount of characters (here I have tried 180), so not all the articles description is on 2 lines, sometimes it's on 3 lines depending on the words length. And if you are on a smaller viewport the text will take even more lines. But at least it does not take seconds to display :)

**Reply by mkrtchian** [reply]
By the way I just tried with 10x more articles by duplicating them on `/articles` page with this code :

```javascript
for(const chunk of chunks) {
    const chunkedArticles = await get<Article[]>('/article/bySlug?slugs[]=' + chunk.join('&slugs[]='));
    for(let i=0; i<10; i++) {
        articles.push(...chunkedArticles);
    }
}
```

On the client it's a bit slower than other pages, but it still takes a reasonable amount of time to load (< 1 second).

With 100x more articles though it takes more time (~3 to 5 seconds) to load the page, so if you have more than 500 articles, you should probably do some sort of pagination, or load them little by little while scrolling.

**Reply by michaelketzer** [reply]
Thanks for reaching out and pointing that issue. I will check it out.
Maybe i can simply use your component with the CSS line-clamp, that should work out very well.

Yea a pagination was not required, yet, but i will definitely add it in future. Thanks a lot.
